### PR TITLE
Build checks catalog

### PR DESCRIPTION
**Pull request by arbulu89**
Implement the checks catalog built process.

This is what we have:

In the Ansible code:
- Update the ansible code to create the catalog for all the available providers. It looks in the `ansible/vars` folder for the different entries, and based on the folders there, it will create a catalog entry for each of the items. So the catalog now is compose like a map where the key is the provider name, and it contains the catalog (the same we were using now). 
- Besides this, the catalog is stored in a file (provider in `CATALOG_DESTINATION` instead of posted to the trento web server

Rest of the code:
- Restructure the App creation to make it testeable with the Dependencies usage.
- Transform the `Runner` struct to a service like struct. To make the mocks possible
- Add `/api/ready` and `/api/catalog` GET requests
- Build the catalog on boot up. The catalog later on is stored on memory, in the runner service. During the catalog creation the `/api/ready` endpoints returns `{"ready": false}` and it is set to true once the catalog creation is done
- The catalog build process is like:
1. Start ansible process
2. Ansible creates the catalog and dump it to a file
3. Golang reads the file and loads it to a variable

The `/api/catalog` would return something like this:
image

PD: If at some point, we want to send the catalog creation as an event, we could use the future webhooks logic, and sent the created catalog once it is built

PD2: By now, all the golang code goes in the same folder, as I didn't want to make more fuzz of it. We can change in the next PR if needed

### CODE REVIEW

File: runner/catalog_api.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package runner
+
+import (
+	"github.com/gin-gonic/gin"
+)
+
+func CatalogHandler(runnerService RunnerService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		c.JSON(200, runnerService.GetCatalog())
```
Comment (arbulu89) [inline runner/catalog_api.go:9]: This returns always `200` state, even though you request the catalog and it is not ready.
We can modify it of course

File: runner/app.go [hunk 2]
```diff
@@ -77,15 +91,27 @@ func (a *App) Start(ctx context.Context) error {
...
+		log.Infof("Starting runner....")
+		g.Go(func() error {
+			err := a.deps.runnerService.Start(ctx)
+			if err != nil {
+				return err
+			}
+			return nil
+		})
+	*/
+
```
Comment (dottorblaster) [inline runner/app.go:114]: Can we delete this?

**Review by dottorblaster — CHANGES_REQUESTED** [review]
One nitpick then we can go

File: runner/app.go (hunk 2 above)
Comment (arbulu89) [reply runner/app.go:114]: Yes, of course

Comment (arbulu89) [reply runner/app.go]: Done: https://github.com/trento-project/runner/pull/3/commits/cbcd45d0550fc2997262ceda1a8005a35ccabfca

**Review by nelsonkopliku — APPROVED** [review]

**Review by dottorblaster — APPROVED** [review]

**Pull request merged**